src/socket/addrinfo.cr

```diff
@@ -121,9 +121,11 @@ class Socket
       when 0
         # success
       when LibC::EAI_NONAME
-        raise Error.new(LibC::EAI_NONAME, "No address found for #{domain}:#{service} over #{protocol}")
+        raise Error.new(ret, "No address found", domain)
+      when LibC::EAI_SOCKTYPE, LibC::EAI_SERVICE
```

Comment:
I'd split those into the different branches:

```cr
when LibC::EAI_SOCKTYPE
  raise Error.new(ret, "The requested socket type is not supported", domain)
when LibC::EAI_SERVICE
  raise Error.new(ret, "The requested service is not available for the requested socket type", domain)
```

Reply:
OK I've split them (and added in the relevant information which I forgot before, heh).  Ready for review, thanks!